images/rootfs.yml.in

```diff
@@ -23,11 +24,17 @@ onboot:
      capabilities:
         - CAP_SYS_ADMIN
         - CAP_NET_ADMIN
+   # If you change the order of storage-init don't forget to
+   # change 002-installer in pkg/mkimage-raw-efi accordingly:
+   # storage-init must follow installer
+   - name: storage-init
+     image: STORAGE_INIT_TAG
+   # kdump goes strictly after storage-init
+   - name: kdump
+     image: KDUMP_TAG
    - name: modprobe
      image: linuxkit/modprobe:v0.5
      command: ["/bin/sh", "-c", "modprobe -a nct6775 w83627hf_wdt hpwdt wlcore_sdio wl18xx br_netfilter dwc3 rk808 rk808-regulator smsc75xx cp210x nicvf tpm_tis_spi rtc_rx8010 gpio_pca953x leds_siemens_ipc127 upboard-fpga pinctrl-upboard leds-upboard xhci_tegra 2>/dev/null || :"]
-   - name: storage-init
-     image: STORAGE_INIT_TAG
```

Comment:
Changing the order (between storage-init and modprobe) we will loose compatibility with booting from USB on Jetson board (as we will not load `xhci_tegra`). Also seems we will not be able to store device models properly during installation (https://github.com/lf-edge/eve/blob/9755360f0479f1b908a9b7eea06c43716c3f1013/pkg/mkimage-raw-efi/install#L481-L483) as we will not load watchdog's modules and possibly some networks drivers (not sure, modprobe line is quite long). So it is not clear to me, can we change the order regardless of intent (common boot or boot after the crash).

Reply:
> I do not quite understand where we get the xhci_tegra module for the x86-64, since it is disabled in the kernel config. Does modprobe fail on x86-64? 

Yes, it is expected to fail as we put here union of modules for all arches. But we still try to load all of them.

> If the only concern is the arm64 - why not make it built-in if we load it unconditionally? This will simplify the modprobe, which is already quite messy.

We have explicit PR to detach the module: https://github.com/lf-edge/eve/pull/2078. @Insei do you remember the reason?

> And why we need network while installation? Do we mount a nbd or something? (I did not find anything). My assumption was simple: all we we need is a block device and a filesystem.

Well, it is not so direct usage, we have a script to compose device model information during an installation (and store it on installation media or in persist). And I can see inside that we want to check [watchdog](https://github.com/lf-edge/eve/blob/9755360f0479f1b908a9b7eea06c43716c3f1013/pkg/debug/spec.sh#L154) presence and do some enumeration of [usb](https://github.com/lf-edge/eve/blob/9755360f0479f1b908a9b7eea06c43716c3f1013/pkg/debug/spec.sh#L203) and [network](https://github.com/lf-edge/eve/blob/9755360f0479f1b908a9b7eea06c43716c3f1013/pkg/debug/spec.sh#L301) devices.

Reply:
Probably we can leave the order, but touch `kernel.modules_disabled` in case of booting after the crash somewhere in early stages?

Reply:
> We have explicit PR to detach the module: #2078. @Insei do you remember the reason?
Ha, I see an explicit description in the PR :)

> Probably we can leave the order, but touch kernel.modules_disabled in case of booting after the crash somewhere in early stages?
Modules are not a problem for the crash kernel, they are all disabled (see 'nomodule' parameter for kexec kernel cmdline): https://github.com/lf-edge/eve/pull/2794/commits/3c98352a641544223c069af1422e90dba246bb31

By changing the order I wanted to expose the assumption: storage-init does not require modules (and should not)!

The only nasty behavior I can foresee if we don't change the order: modprobe fails with EPERM in crash kernel because of the 'nomodule' parameter. Well, not a big deal, but ugly.